jgrapht-ext/src/main/java/org/jgrapht/ext/GraphMLExporter.java

```diff
@@ -22,106 +22,503 @@
+                "Attribute category must be one of node, edge, graph or all");
+        }
+        String nextKey = ATTRIBUTE_KEY_PREFIX + (totalAttributes++);
+        registeredAttributes.put(
+            name,
+            new AttributeDetails(nextKey, category, type, defaultValue));
+    }
+
+    /**
+     * Unregister a GraphML-Attribute
+     * 
+     * @param name the attribute name
+     */
+    public void unregisterAttribute(String name)
+    {
+        if (name == null) {
+            throw new IllegalArgumentException("Attribute name cannot be null");
+        }
+        if (name.equals(vertexLabelAttributeName)
+            || name.equals(edgeWeightAttributeName)
+            || name.equals(edgeLabelAttributeName))
+        {
+            throw new IllegalArgumentException("Reserved attribute name");
+        }
+        registeredAttributes.remove(name);
+    }
+
+    /**
+     * Whether the exporter will print edge weights.
+     *
+     * @return {@code true} if the exporter prints edge weights, {@code false}
+     *         otherwise
+     */
+    public boolean isExportEdgeWeights()
+    {
+        return exportEdgeWeights;
+    }
+
+    /**
+     * Set whether the exporter will print edge weights.
+     *
+     * @param exportEdgeWeights value to set
+     */
+    public void setExportEdgeWeights(boolean exportEdgeWeights)
+    {
+        this.exportEdgeWeights = exportEdgeWeights;
+    }
+
+    /**
+     * Get the attribute name for vertex labels
+     * 
+     * @return the attribute name
+     */
+    public String getVertexLabelAttributeName()
+    {
+        return vertexLabelAttributeName;
+    }
+
+    /**
+     * Set the attribute name to use for vertex labels.
+     * 
+     * @param vertexLabelAttributeName the attribute name
+     */
+    public void setVertexLabelAttributeName(String vertexLabelAttributeName)
+    {
+        if (vertexLabelAttributeName == null) {
+            throw new IllegalArgumentException(
+                "Vertex label attribute name cannot be null");
+        }
+        if (registeredAttributes.containsKey(vertexLabelAttributeName)) {
+            throw new IllegalArgumentException("Reserved attribute name");
+        }
+        this.vertexLabelAttributeName = vertexLabelAttributeName;
+    }
+
+    /**
+     * Get the attribute name for edge labels
+     * 
+     * @return the attribute name
+     */
+    public String getEdgeLabelAttributeName()
+    {
+        return edgeLabelAttributeName;
+    }
+
+    /**
+     * Set the attribute name to use for edge labels.
+     * 
+     * @param edgeLabelAttributeName the attribute name
+     */
+    public void setEdgeLabelAttributeName(String edgeLabelAttributeName)
+    {
+        if (edgeLabelAttributeName == null) {
+            throw new IllegalArgumentException(
+                "Edge label attribute name cannot be null");
+        }
+        if (registeredAttributes.containsKey(edgeLabelAttributeName)) {
```

Comment:
I changed the code to trim additional spaces from the attribute names. 

Additional lowercase is not needed here, as this is the user provided attribute name. In the example below this is the "Vertex Label".

```
<key id="vertex_label_key" for="node" attr.name="Vertex Label" attr.type="string"/>
```